src/integrationTest/java/org/openstreetmap/atlas/generator/slicing/DynamicRelationSlicingTest.java

```diff
@@ -0,0 +1,118 @@
+package org.openstreetmap.atlas.generator.slicing;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openstreetmap.atlas.generator.AtlasGeneratorHelper;
+import org.openstreetmap.atlas.generator.tools.caching.HadoopAtlasFileCache;
+import org.openstreetmap.atlas.generator.tools.streaming.ResourceFileSystem;
+import org.openstreetmap.atlas.geography.Polygon;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.raw.slicing.RawAtlasCountrySlicer;
+import org.openstreetmap.atlas.geography.boundary.CountryBoundaryMap;
+import org.openstreetmap.atlas.geography.sharding.CountryShard;
+import org.openstreetmap.atlas.geography.sharding.DynamicTileSharding;
+import org.openstreetmap.atlas.geography.sharding.Shard;
+import org.openstreetmap.atlas.geography.sharding.Sharding;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.InputStreamResource;
+
+/**
+ * @author james-gage
+ */
+public class DynamicRelationSlicingTest
+{
+    public static final String ABC_100 = "resource://test/atlas/ABC/ABC_1-0-0.atlas";
+    public static final String ABC_101 = "resource://test/atlas/ABC/ABC_1-0-1.atlas";
+    public static final String ABC_110 = "resource://test/atlas/ABC/ABC_1-1-0.atlas";
+    public static final String ABC_111 = "resource://test/atlas/ABC/ABC_1-1-1.atlas";
+    public static final String DEF_101 = "resource://test/atlas/DEF/DEF_1-0-1.atlas";
+    public static final String DEF_111 = "resource://test/atlas/DEF/DEF_1-1-1.atlas";
+
+    static
+    {
+        addResource(ABC_100, "ABC_1-0-0.atlas");
+        addResource(ABC_101, "ABC_1-0-1.atlas");
+        addResource(ABC_110, "ABC_1-1-0.atlas");
+        addResource(ABC_111, "ABC_1-1-1.atlas");
+        addResource(DEF_101, "DEF_1-0-1.atlas");
+        addResource(DEF_111, "DEF_1-1-1.atlas");
+    }
+
+    private static void addResource(final String path, final String name)
+    {
+        ResourceFileSystem.addResource(path, new InputStreamResource(
```

Comment:
Maybe store the Atlas files as text (to avoid binary) and then load them into `PackedAtlas` here